proto/cacheclient.proto

```diff
@@ -48,3 +51,34 @@ message _SetResponse {
   ECacheResult result = 1;
   string message = 2;
 }
+
+message _HashGetRequest {
```

Comment:
to paraphrase kenny's point, he pointed out that there are many different names for this data structure in different tools/languages.  hash, associative array, map, object, dictionary.  And we shouldn't consider ourselves locked into one noun that our competitors use.

I think "map" and "object" are too overloaded.  I think "associative array" is too wordy.

I like "dict" and "hash" the best.  Personally I would tie-break between those two in favor of "dict" simply because the word "hash" is also kind of overloaded in programming.  "dict" doesn't really have any other meanings.

So I'd probably vote for `Dict*`, but `Hash*` would be my second choice.

Reply:
Thanks Chris. If we're open to this I prefer Dict too because we also say hash as a verb. Dicts are not verbs though and imo are a better naming candidate for this new noun in our api surface.

Reply:
> did we bottom out on `Hash` as the noun here? Kenny had left some comments somewhere that I thought were interesting on this topic.

No. I wanted to field this for broader discussion.

I agree with Kenny's points that "hash" is not the best. A hash could be a hash table, a hash function, the image of a hash function, a hashmap, ... etc. Also hashing is a particular way to implement a dictionary. I don't see anyone clamoring for the name "RedBlackTreeMap".

`map` could be the `map` higher-order function. `object` is vague and `associative array` is too long like you said. I think Dict* could be good.

Reply:
Renamed Hash->Dict.